src/main/java/edu/kit/compiler/codegen/pattern/DivisionPattern.java

```diff
@@ -27,10 +28,12 @@ public InstructionMatch match(Node node, MatcherState matcher) {
 
             var lhs = left.match(node.getPred(1), matcher);
             var rhs = right.match(node.getPred(2), matcher);
-            var destination = matcher.getNewRegister();
+
+            // todo is hard coding the register size the best choice here?
+            var targetRegister = matcher.getNewRegister(RegisterSize.QUAD);
```

Comment:
This is probably a mistake; unlike the inputs, the target register of a `div` needs to have size `DOUBLE`. The conversion from `QUAD` to `DOUBLE` happens implicitly when accessing the register (see next comment)